monthly-contributions/src/main/scala/com/gu/support/workers/lambdas/CreatePaymentMethod.scala

```diff
@@ -53,22 +47,40 @@ class CreatePaymentMethod(servicesProvider: ServiceProvider = ServiceProvider)
       state.acquisitionData
     )
 
-  def createStripePaymentMethod(stripe: StripePaymentFields, stripeService: StripeService): Future[CreditCardReferenceTransaction] =
+  def createStripePaymentMethod(stripe: StripePaymentFields, stripeService: StripeService): Future[CreditCardReferenceTransaction] = {
     stripeService
       .createCustomer(stripe.userId, stripe.stripeToken)
       .map { stripeCustomer =>
         val card = stripeCustomer.card
         CreditCardReferenceTransaction(card.id, stripeCustomer.id, card.last4,
           CountryGroup.countryByCode(card.country), card.exp_month, card.exp_year, card.zuoraCardType)
       }
+  }
 
-  def createPayPalPaymentMethod(payPal: PayPalPaymentFields, payPalService: PayPalService): Future[PayPalReferenceTransaction] =
+  def createPayPalPaymentMethod(payPal: PayPalPaymentFields, payPalService: PayPalService): Future[PayPalReferenceTransaction] = {
     payPalService
       .retrieveEmail(payPal.baid)
       .map(PayPalReferenceTransaction(payPal.baid, _))
+  }
+
+  private def putStartup(paymentType: Either[StripePaymentFields, PayPalPaymentFields]) =
```

Comment:
You can refactor out this code duplication by replacing `putStartup` & `putPaymentMethodCreated` with:

      private def putMetric(paymentType: Either[StripePaymentFields, PayPalPaymentFields], f: String => Future[Unit]) =
        if (paymentType.isLeft)
          f("stripe")
        else
          f("paypal")

And then call this method above with `putMetric(state.paymentFields, startupMetric)` & `putMetric(state.paymentFields, paymentMethodCreatedMetric)`

Reply:
nice!